packages/cascara/src/ui/Table/Table.test.js

```diff
@@ -332,88 +351,95 @@ describe('Table', () => {
 
       render(
         <Table
+          actions={actions}
           data={data}
           dataConfig={dataConfig}
           onAction={onAction}
           uniqueIdAttribute={'eid'}
         />
       );
 
-      const editButton = screen.getAllByTestId('edit.start');
+      const editButton = screen.getAllByRole('button', {
+        name: 'Edit',
+      });
 
       // Enter edit mode
-      fireEvent(
-        editButton[0],
-        new MouseEvent('click', {
-          bubbles: true,
-          cancelable: true,
-        })
+      // fireEvent(
+      //   editButton[0],
+      //   new MouseEvent('click', {
+      //     bubbles: true,
+      //     cancelable: true,
+      //   })
+      // );
+      userEvent.click(editButton[0]);
+
+      await screen.findAllByRole('button', {
+        name: 'Cancel',
+      });
+
+      const emailTextbox = screen.getByRole('textbox', { name: 'Email' });
```

Comment:
@brianespinosa we are not able to use the `byRole` queryset when testing, this is because all components appear by `role` (textbox) but the `name` resolves to an `empty string`. So, in the test, we cannot specifically-select `email`.

It would be nice if we could add the `aria-label` to all modules, just like we specify `labels` for them. In this case, it would have helped a lot because we can't query by the label.
![image](https://user-images.githubusercontent.com/13924868/110900538-83214d80-82c8-11eb-94ce-6b27542b5895.png)

That's why the test are failing. This would be solved if in [DataEmail]() we added the `aria-label` prop... Because `reakit` does not add one.

Reply:
@lordpixel are we are already spreading {...rest} props to the modules from the object? Why not just define `data-label` to the object that gets mapped to each module?

Reply:
It looks like maybe this comment is old. I looked at the API and based on that the [accessible name](https://www.tpgi.com/what-is-an-accessible-name/) guidelines it would appear that the `label` is what `{ name: X}` would resolve to.

Looks like your tests are updated using that now.

I love that RTL is using accessibility to drive this test selector. <3

Reply:
Okay, I see what is going on. The reason `name` is empty is because the input has an `id` and it is linked by a `<label>` that has a `for` prop matching that `id`. In the case we are not passing a label, we are still rendering the `<label>` element in the DOM. There is no value string set inside that so `name` will select as empty.

```
<label
  for="Email"
>
  // This would be where the label string would normally appear which I think would show up with
  // screen.getByRole('textbox', { name: 'Email' })
  <input
    class="Input"
    data-testid="email"
    id="Email"
    name="email"
    type="email"
    value="Hayden.Zieme@espressive.com"
  />
</label>
```

I think this is actually raising an accessibility issue because this is what a screen reader would see for that input and it would be not having a label.

I think you keep the test selector exactly the same. The test can drive our code changes we need to make. The test is correct. The components are failing!

- We should actually NOT render a `<label>` tag if the component is set to `isLabeled: false`. We do still need that node for styling purposes, but I think the CSS will work if we just render a `<span>` instead. (Both label and span have the same default display inline without any CSS).

- We should be setting an `aria-label` on the input with the name instead, but ONLY if the `isLabeled` is set to true. This way we do not render the label in the UI as a tag, but we still label that input directly which will not break screen readers.

Reply:
Nice, thanks for the comfirmation, working on the changes. 🙌

Reply:
@brianespinosa tests are passing now, I added `aria-label` to the `column` [definition](https://github.com/Espressive/cascara/pull/192/files#diff-e57c8635c338b80585061dff8863053d3538c48fba44d62b5594f072d1b58289R84) in `dataConfig.display`.

Since we are ...spreading, the prop reaches the DOM elemen and that makes the test green. This goes to show that although we are not setting those internally, developers can do it if they need to.

I will create an FDS task for implementing the accessibility improvements.

Reply:
[FDS-170](https://espressive.atlassian.net/browse/FDS-170) has been created for this.